### PR TITLE
feat: add signature computation/verification features

### LINKED ISSUE #28: Add "prepareComputeSignature" method for Card and Sam transactions

### LINKED ISSUE #29: Add "preparePsoVerifySignature" method for Card and Sam transactions

### PR DESCRIPTION
**Pull request by andrei-cristea**
Closes #28, closes #29

### CODE REVIEW

File: src/main/java/org/calypsonet/terminal/calypso/transaction/SignatureVerificationData.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+  /**
+   * Returns the result of the signature verification process by indicating if the signature is
+   * valid or not.
+   *
+   * @return True if the signature is valid.
+   * @throws IllegalStateException If the command has not yet been processed.
+   * @since 1.2.0
+   */
+  boolean isSignatureValid();
```
Comment (pterr) [inline src/main/java/org/calypsonet/terminal/calypso/transaction/SignatureVerificationData.java:99]: During the data signature verification, if the traceability data is present, AND if a SamRevocationServiceSpi is defined => then the validity of the SAM which has generated the signature could be automatically checked.
If the SAM isn't valid an exception could be issued.

File: src/main/java/org/calypsonet/terminal/calypso/transaction/SignatureVerificationData.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+   * Returns the full (4 bytes) or partial (3 LSBytes) SAM serial number written in the traceability
+   * information if the check of the "SAM traceability" data has been enabled.
+   *
+   * @return A byte array of 3 or 4 bytes.
+   * @throws IllegalStateException If the command has not yet been processed or if the check of the
+   *     "SAM traceability" data is disabled.
+   * @see #checkSamTraceabilityData(int, boolean, boolean)
+   * @since 1.2.0
+   */
+  byte[] getSamTraceabilitySerialNumber();
```
Comment (pterr) [inline src/main/java/org/calypsonet/terminal/calypso/transaction/SignatureVerificationData.java:111]: For me, if the SamRevocationServiceSpi is automatically processed then:
 - getSamTraceabilitySerialNumber()
 - getSamTraceabilityKeyCounter()
are low-level command (not necessary for the terminal application)

File: src/main/java/org/calypsonet/terminal/calypso/transaction/SignatureVerificationData.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+   *
+   * <p>Note that after a reset of the SAM, "PSO Verify Signature" commands being in "Busy" mode
+   * fail with the busy status until the end of the busy mode duration.
+   *
+   * <p>By default, the "Busy" mode is enabled.
+   *
+   * @return The current instance.
+   * @since 1.2.0
+   */
+  SignatureVerificationData disableBusyMode();
```
Comment (pterr) [inline src/main/java/org/calypsonet/terminal/calypso/transaction/SignatureVerificationData.java:77]: To remove, the busy mode is defined only at the computation of the signature

File: src/main/java/org/calypsonet/terminal/calypso/transaction/SamSecuritySetting.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+ * SPDX-License-Identifier: EPL-2.0
+ ************************************************************************************** */
+package org.calypsonet.terminal.calypso.transaction;
+
+/**
+ * Data to manage the security operations of a Calypso SAM transaction.
+ *
+ * @since 1.2.0
+ */
+public interface SamSecuritySetting extends CommonSecuritySetting<SamSecuritySetting> {}
```
Comment (pterr) [inline src/main/java/org/calypsonet/terminal/calypso/transaction/SamSecuritySetting.java:19]: Here, we will certainely define in the future a method like:

SamSecuritySetting setControlSamResource (CardReader controlSamReader, CalypsoSam controlSam)

**Review by pterr — CHANGES_REQUESTED** [review]

File: src/main/java/org/calypsonet/terminal/calypso/transaction/SignatureComputationData.java [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+  /**
+   * Requires to perform a key diversification before signing using the provided diversifier.
+   *
+   * <p>By default, there will be no new diversification of the key.
+   *
+   * @param keyDiversifier The diversifier to be used (8 bytes long).
+   * @return The current instance.
+   * @since 1.2.0
+   */
+  SignatureComputationData useKeyDiversifier(byte[] keyDiversifier);
```
[the remark this answers is not in the document]
Comment (andrei-cristea) [reply src/main/java/org/calypsonet/terminal/calypso/transaction/SignatureComputationData.java:90]: The documentation has been updated.

File: src/main/java/org/calypsonet/terminal/calypso/transaction/SignatureVerificationData.java [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+   * Requires to perform a key diversification before verifying the signature using the provided
+   * diversifier.
+   *
+   * <p>By default, there will be no new diversification of the key.
+   *
+   * @param keyDiversifier The diversifier to be used (8 bytes long).
+   * @return The current instance.
+   * @since 1.2.0
+   */
+  SignatureVerificationData useKeyDiversifier(byte[] keyDiversifier);
```
[the remark this answers is not in the document]
Comment (andrei-cristea) [reply src/main/java/org/calypsonet/terminal/calypso/transaction/SignatureVerificationData.java:89]: The documentation has been updated.

**Review by pterr — APPROVED** [review]

**Pull request merged**